websites/H/Hoyolab/presence.ts

```diff
@@ -0,0 +1,110 @@
+const presence = new Presence({
+  clientId: "836534947170353173"
+}),
+  browsingStamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+  const data: PresenceData = {
+    largeImageKey: "logo"
+  },
+    pathname = document.location.pathname;
+    if (pathname === "/genshin/"){
+      data.details = "Viewing the Homepage";
+      data.startTimestamp = browsingStamp;
+    }
+    else if(pathname.endsWith("/home/1") && document.location.search.substr(6) == "create"){
+      data.details = "Viewing tavern page";
+      data.state = "New";
+      data.startTimestamp = browsingStamp;
+    }
+    else if(pathname.endsWith("/home/1") && document.location.search.substr(6) == "reply"){
+      data.details = "Viewing tavern page";
+      data.state = "New replies";
+      data.startTimestamp = browsingStamp;
+    }
+    else if(pathname.endsWith("/home/1") && document.location.search.substr(6) == "2"){
+      data.details = "Viewing tavern page";
+      data.state = "Featured";
+      data.startTimestamp = browsingStamp;
+    }
+    else if(pathname.endsWith("/home/1")){
+      data.details = "Viewing tavern page";
+      data.state = "Hot";
+      data.startTimestamp = browsingStamp;
+    }
```

Comment:
you could merge it and like do if inside if instead of doing same thing 3 times.
example:
```javascript
if(pathname.endsWith("/home/1") {
   switch(document.location?.search?.substr(6)) {
       case "create": /* code */
       case "reply": /* code */
       case "2": /* code */
       case null: /* code */
   }
}
```

Reply:
ooh, I didn't think of switch, I'll do it for all those below